src/hdmf/data_utils.py

```diff
@@ -127,6 +128,19 @@ def maxshape(self):
         raise NotImplementedError("maxshape not implemented for derived class")
 
 
+# TODO: remove this function
+def _temporary_prod(iterable):
+    """
+    We want to use math.prod but it is only available in Python >= 3.8 versions.
+
+    Once 3.7 support is dropped, this function can be removed
+    """
+    result = 1
+    for x in iterable:
+        result *= x
+    return result
```

Comment:
Since this function is only use by ``GenericDataChunkIterator`` would it make sense to make this a private classmethod of ``GenericDataChunkIterator`` rather than exposing this on the module level?

End of life for Python 3.7 seems to be in ~7 months (27 Jun 2023). @CodyCBakerPhD can you create an issue ticket to make sure we have record/reminder that this function should be removed in the future.

Reply:
A one-line alternative for `` math.prod `` would be to use the following  ``functools.reduce(operator.mul, shape, 1)``

Reply:
@rly what are your thoughts regarding the ``_temporary_prod``.

Reply:
I'll just switch to `functools.reduce(operator.mul, iterable, 1)` for now and leave notes and tickets that it can be simplified later on

Reply:
Reminder filed: https://github.com/hdmf-dev/hdmf/issues/785

Reply:
Perfect. Thanks!